lib/symboldatabase.cpp

```diff
@@ -917,6 +893,32 @@ void SymbolDatabase::createSymbolDatabaseVariableInfo()
     }
 }
 
+void SymbolDatabase::createSymbolDatabaseCopyAndMoveConstructors()
+{
+    // fill in class and struct copy/move constructors
+    for (std::list<Scope>::iterator scope = scopeList.begin(); scope != scopeList.end(); ++scope) {
+        if (scope->isClassOrStruct()) {
+            std::list<Function>::iterator func;
+
+            for (func = scope->functionList.begin(); func != scope->functionList.end(); ++func) {
+                if (func->isConstructor() && func->minArgCount() == 1) {
+                    const Variable* firstArg = func->getArgumentVar(0);
+                    if (firstArg->type() == scope->definedType) {
+                        if (firstArg->isRValueReference())
+                            func->type = Function::eMoveConstructor;
+                        else if (firstArg->isReference())
+                            func->type = Function::eCopyConstructor;
+                    }
+
+                    if (func->type == Function::eCopyConstructor ||
+                        func->type == Function::eMoveConstructor)
+                        scope->numCopyOrMoveConstructors++;
```

Comment:
how about zeroing numCopyOrMoveConstructors in this method. Can numCopyOrMoveConstructors be incremented only in the above condition?

Reply:
`numCopyOrMoveConstructors` is only incremented at this point, at it's initialized to zero in the constructor of `Scope`.
I don't think zeroing it is necessary, but I don't have a strong opinion against it either.